crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -301,6 +301,7 @@ public Response update(UpdateRequest req) {
                         updateResponse.setNumMatched(0);
                     }
                 }
+                ctx.getHookManager().ensureQueuedHooksSizeNotTooLarge(factory.getMaxResultSetSizeB(), factory.getWarnResultSetSizeB(), req);
```

Comment:
Take a look at Mediator#setResponseResults:840-843. If no hooks were queued for processing, it will use the thresholds in the Response. Mediator#setResponseResults is called for inserts, saves and deletes (all writes). Find doesn't do hooks at all, so it's always using Response implementation.